### PR TITLE
UI Changes for Configuring Proxy-Cache Speed Limit

### LINKED ISSUE #20811: proxy-cache bandwidth limit UI change
Issue https://github.com/goharbor/harbor/issues/20589 will provide the feature of bandwidth limit for proxy-cache. This current issue is to discuss the UI change of configuring proxy-cache speed limit:

- configuring proxy-cache speed limit in the proxy-cache creation page
Screenshot 2024-08-06 at 1 54 20 PM

- configuring proxy-cache speed limit in the project configuration page (like: auto_scan)
Screenshot 2024-08-06 at 1 56 17 PM

### PR DESCRIPTION
**Pull request by kunal-511**
Thank you for contributing to Harbor!

proxy-cache bandwidth limit UI change
# Comprehensive Summary of your change
This pull request addresses the UI changes required for configuring the proxy-cache speed limit, as outlined in Issue #20589. The changes include the addition of a bandwidth field for setting the speed limit in both the proxy-cache creation page and the project configuration page
# Issue being fixed
Fixes #20811 

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. "release-note/ignore-for-release, release-note/new-feature, release-note/update, release-note/enhancement, release-note/community, release-note/breaking-change, release-note/docs, release-note/infra, release-note/deprecation"
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [ ] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

File: src/portal/src/i18n/lang/en-us-lang.json [hunk 1]
```diff
@@ -246,6 +246,7 @@
         "TOGGLED_SUCCESS": "Toggled project successfully.",
         "FAILED_TO_DELETE_PROJECT": "Project contains repositories or replication rules or helm-charts cannot be deleted.",
         "INLINE_HELP_PUBLIC": "When a project is set to public, anyone has read permission to the repositories under this project, and the user does not need to run \"docker login\" before pulling images under this project.",
+        "PROXY_CACHE_BANDWIDTH":"Set the maximum network bandwidth to pull image from upstream for proxy-cache. For unlimited bandwidth, please enter -1. ",
```
Comment (Vad1mo) [inline src/portal/src/i18n/lang/en-us-lang.json:249]: this string needs to be added to all language files. (Put it in the english version), for later translation.

 Its than easier to pick it up by others and translate

**Review by Vad1mo — CHANGES_REQUESTED** [review]

File: src/portal/src/i18n/lang/en-us-lang.json (hunk 1 above)
Comment (kunal-511) [reply src/portal/src/i18n/lang/en-us-lang.json:249]: Added this line to all the language files

**Comment by stonezdj** [comment]
I verify the UI, the payload is wrong:
```
{"project_name":"dockerhub5","metadata":{"public":"true"},"storage_limit":-1,"registry_id":1}
```
It should be in this format
```
{"project_name":"dockerhub5","metadata":{"public":"true", "storage_limit":-1},"registry_id":1}
```

**Comment by stonezdj** [comment]
Screenshot 2024-08-30 at 15 00 49
The color in the warning message is different from the existing color.

**Comment by stonezdj** [comment]
@kunal-511  could please refer to this PR https://github.com/stonezdj/harbor/commit/e39f73edd617afb7a933a2c4a77847d792964333 to update your code.

**Comment by kunal-511** [comment]
@stonezdj  Updated my PR

**Comment by chlins** [comment]
@kunal-511 Please resolve the code conflicts.

**Comment by kunal-511** [comment]
> @kunal-511 Please resolve the code conflicts.

Resolved the code conflicts

**Comment by kunal-511** [comment]
> @kunal-511 could please refer to this PR [stonezdj@e39f73e](https://github.com/stonezdj/harbor/commit/e39f73edd617afb7a933a2c4a77847d792964333) to update your code.

need to update other language json files ??

**Comment by stonezdj** [comment]
> @stonezdj Updated my PR

Please update the language json file to add this line.
```
        "BANDWIDTH": "Bandwidth",
        "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
```

**Comment by kunal-511** [comment]
> > @stonezdj Updated my PR
> 
> Please update the language json file to add this line.
> 
> ```
>         "BANDWIDTH": "Bandwidth",
>         "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
> ```

Need to add this to all language json files?

**Comment by stonezdj** [comment]
> > > @stonezdj Updated my PR
> > 
> > 
> > Please update the language json file to add this line.
> > ```
> >         "BANDWIDTH": "Bandwidth",
> >         "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
> > ```
> 
> Need to add this to all language json files?

Yes

File: src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.scss [hunk 2]
```diff
@@ -1,6 +1,9 @@
 #severity-blk div {
   display: inline-block;
 }
+.mr-10 {
```
Comment (stonezdj) [inline src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.scss:4]: The current UI change is only limit to create project, why change the CSS of the project policy config?

**Comment by stonezdj** [comment]
Please squash all commit into one @kunal-511

**Comment by kunal-511** [comment]
> Please squash all commit into one @kunal-511

Yes I will do this

File: src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.scss (hunk 2 above)
Comment (kunal-511) [reply src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.scss:4]: > The current UI change is only limit to create project, why change the CSS of the project policy config?

It was initially told by @zyyw that we need to make changes to both the create project and also to the  project policy-config page

File: src/portal/src/app/base/left-side-nav/projects/create-project/create-project.component.ts [hunk 3]
```diff
@@ -290,18 +308,53 @@ export class CreateProjectComponent
...
+    // **Added method for validating bandwidth input**
+    validateBandwidth(): void {
+        const value = Number(this.speedLimit);
+        if (
+            isNaN(value) ||
+            (!Number.isInteger(value) && value !== -1) ||
+            (value <= 0 && value !== -1)
+        ) {
+            this.bandwidthError =
+                'Please enter -1 or an integer  greater than 0.';
```
Comment (xuelichao) [inline src/portal/src/app/base/left-side-nav/projects/create-project/create-project.component.ts:328]: Remove extra white space between word 'integer' and 'greater'

File: src/portal/src/i18n/lang/zh-tw-lang.json [hunk 4]
```diff
@@ -245,12 +245,15 @@
...
     "INLINE_HELP_PUBLIC": "當專案設為公開時，任何人都可讀取此專案下的儲存庫，使用者不需執行 \"docker login\" 即可拉取此專案下的映像檔。",
+    "PROXY_CACHE_BANDWIDTH":"Set the maximum network bandwidth to pull image from upstream for proxy-cache. For unlimited bandwidth, please enter -1. ",
     "OF": "共計",
     "COUNT_QUOTA": "數量配額",
     "STORAGE_QUOTA": "儲存配額限制",
     "COUNT_QUOTA_TIP": "請輸入介於 '1' 和 '100,000,000' 之間的整數，'-1' 表示無限制。",
     "STORAGE_QUOTA_TIP": "儲存配額上限只接受整數，上限為 '1024TB'。輸入 '-1' 表示無限配額。",
     "QUOTA_UNLIMIT_TIP": "專案所能使用的最大邏輯空間。輸入 '-1' 表示無限制配額。",
+    "BANDWIDTH": "Bandwidth",
+        "SPEED_LIMIT_TIP": "Please enter -1 or an integer  greater than 0. ",
```
Comment (xuelichao) [inline src/portal/src/i18n/lang/zh-tw-lang.json:256]: remove the extra indent

File: src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.html [hunk 5]
```diff
@@ -16,6 +16,42 @@
...
             </clr-control-helper>
         </clr-checkbox-container>
+        <section *ngIf="isProxyCacheProject">
+            <div class="clr-form-control margin-top-05">
+                <label for="" class="clr-control-label required"
+                    >ProxyCache Bandwidth</label
+                >
+                <div class="flex">
+                    <input
+                        class="clr-input width-164 mr-10"
```
Comment (xuelichao) [inline src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.html:26]: why does the [(ngModel)] not configured for the input element?

File: src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.html [hunk 6]
```diff
@@ -16,6 +16,42 @@
...
+                        type="text"
+                        placeholder="ProxyCache Bandwidth"
+                        autocomplete="off" />
+                    <div class="clr-select-wrapper mr-10">
+                        <select name="" id="">
+                            <option value="kbps" selected>kbps</option>
+                            <option value="Mbps">Mbps</option>
```
Comment (xuelichao) [inline src/portal/src/app/base/project/project-config/project-policy-config/project-policy-config.component.html:33]: [(ngModel)] is not configured for select. why?

**Comment by kunal-511** [comment]
@stonezdj I have created a new PR with single commit and also addressed above issues. Please have a look [#20946](https://github.com/goharbor/harbor/pull/20946)

**Pull request closed without merging**